indexer/src/main/java/au/org/aodn/esindexer/service/AodnDiscoveryParameterVocabService.java

```diff
@@ -0,0 +1,70 @@
+package au.org.aodn.esindexer.service;
+
+import au.org.aodn.esindexer.abstracts.OgcApiRequestEntityCreator;
+import au.org.aodn.esindexer.utils.CacheArdcVocabsUtils;
+import au.org.aodn.stac.model.ConceptModel;
+import au.org.aodn.stac.model.ThemesModel;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class AodnDiscoveryParameterVocabService {
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    OgcApiRequestEntityCreator ogcApiRequestEntityCreator;
+
+    @Autowired
+    CacheArdcVocabsUtils cacheArdcVocabsUtils;
+
+    protected boolean themesMatchConcept(List<ThemesModel> themes, ConceptModel concept) {
+        for (ThemesModel theme : themes) {
+            for (ConceptModel themeConcept : theme.getConcepts()) {
+                /*
+                comparing by hashcode (id and url) of the concept object
```

Comment:
Comment not correct, it is comparing by values.

Reply:
ah true, will delete this old comment of when I attempted to do hash comparison, no longer relevant with the use of override `equals` method.